core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -212,6 +213,16 @@ public synchronized void connect() throws AlluxioStatusException {
             getServiceName(), retryPolicy.getAttemptCount(), e.toString());
         continue;
       }
+      if (mAddress.isUnresolved()) {
+        // Sometimes the acquired addressed wasn't resolved, retry resolving before
+        // using it to connect.
+        LOG.info("Retry resolving address {}", mAddress);
+        // Creates a new InetSocketAddress to force resolving the hostname again.
+        mAddress = new InetSocketAddress(mAddress.getHostName(), mAddress.getPort());
+        if (mAddress.isUnresolved()) {
+          LOG.warn("Failed to resolve address on retry {}", mAddress);
```

Comment:
Maybe a little sleep for giving network stack a chance?

Reply:
The retry policy will make sure the client sleeps before trying another resolution.